### PR TITLE
SDIT-1062: ✨ Schedule in deletion of old difference data

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: helm_deploy/prisoner-offender-search/templates/elasticsearch-daily-snapshot.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ data:
 {{ .Files.Get "snapshot.sh" | indent 4 }}
 
 ---
-apiVersion: batch/v1beta1
+apiVersion: batch/v1
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/templates/elasticsearch-daily-snapshot.yaml:13]: `v1beta1` deprecated - new version is just `v1`

File: helm_deploy/prisoner-offender-search/templates/remove-old-differences-data-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "app.fullname" . }}-index-housekeeping
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  schedule: "45 22 * * *"
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/templates/remove-old-differences-data-cronjob.yaml:8]: run once at day at 22:45

File: helm_deploy/prisoner-offender-search/values.yaml [hunk 3]
```diff
@@ -36,6 +36,10 @@ generic-service:
           deny all;
           return 401;
         }
+        location /prisoner-differences/delete {
```
Comment (petergphillips) [inline helm_deploy/prisoner-offender-search/values.yaml:39]: endpoint needs to be called from cronjob so ingress protected so can only be called from inside namespace

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerDifferencesResource.kt [hunk 4]
```diff
@@ -63,11 +65,16 @@ class PrisonerDifferencesResource(private val prisonerDifferencesService: Prison
...
-    @Parameter(description = "Report on differences that have been generated. Defaults to the last 24 hours", example = "2023-01-02T02:23:45")
+    @Parameter(description = "Report on differences that have been generated. Defaults to the last one day", example = "2023-01-02T02:23:45")
     from: Instant?,
     to: Instant?,
   ): List<PrisonerDifferences> =
-    prisonerDifferencesService.retrieveDifferences(from ?: Instant.now().minusSeconds(60 * 60 * 24), to ?: Instant.now())
+    prisonerDifferencesService.retrieveDifferences(from ?: Instant.now().minus(1, ChronoUnit.DAYS), to ?: Instant.now())
+
+  @Hidden
+  @DeleteMapping("/delete")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerDifferencesResource.kt:78]: no authentication on endpoint

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**